main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -261,7 +261,15 @@ object GenExpression {
         visitor.visitFieldInsn(PUTFIELD, className, s"clo$i", JvmOps.getClosureAbstractClassType(m.clo.tpe).toDescriptor)
       }
 
+    case Expr.Instanceof(exp, name, _, loc) =>
+      addSourceLine(visitor, loc)
+      val className = JvmName(ca.uwaterloo.flix.language.phase.jvm.JvmName.RootPackage, name).toInternalName
+      println(className);
+      compileExpression(exp, visitor, currentClass, lenv0, entryPoint)
+      visitor.visitTypeInsn(INSTANCEOF, className.toString)
```

Comment:
@JonathanStarup Can you help with the right incantations?

Reply:
Thanks Jonathan - `instanceof` expressions are now working due to using `Class[_]` and `asm.Type.getInternalName`.